Add Parameter to Override ApiEndpoint

stian3555 opened the pull request:
With SendGrid now supporting an EU endpoint, I would think it's beneficial to introduce a parameter that allows users to override the default API endpoint. By retaining the current endpoint as the default value, this change will not be breaking for existing users.

Alternatively, a region parameter could be added, with predefined endpoints for each region. However, this approach would require updates if SendGrid introduces additional regions.

For more details, refer to SendGrid’s EU endpoint announcement.
https://sendgrid.com/en-us/blog/Configurable-Sending-in-EU-Now-in-GA

stian3555 committed:
Allow override ApiEndpoint

Ba4bes commented:
Looks good! Two small details:
- Could you move the new parameter to line 187? I think it makes sense if it is right next to the $token parameter
- Could you update the command-help with a small description like you did with the other documents? So add it in the comment in line 93.

I will try to find some time to do some testing just-to-be-sure and add it to the gallery

stian3555 committed:
Change position of new parameter to be below the token, updated help xml

stian3555 commented:
@Ba4bes I have now updated the position of the parameter and updated the command-help.  

This can result in a breaking change if someone is using the positional parameters, but I guess most people use named parameters.

I also now noticed that there is a discrepancy between Send-PSSendGridMail.md and PSSendGrid-help.xml in terms of order of positional parameters, Send-PSSendGridMail.md seems to be correct.

Ba4bes committed:
Update api endpoint to be a URI.

parameter will not work if it is a string[]. Updated documentation to come from Comment-help

Ba4bes commented:
The documentation files are automatically generated from the comment-help. I have fixed it in a new commit. 
Also changed the type to URI, otherwise it will not work.
I did all tests and they work, so will upload to the Gallery as soon as I can.

Merging this one now. Thank you for the suggestion and your help!

stian3555 commented:
@Ba4bes I noticed the module has been updated in the gallery. Thank you so much for addressing this so promptly, it saves me a lot of time from having to create a new solution myself.

I also want to express my gratitude for the fantastic community effort in developing this module. Your work is greatly appreciated! 😊
